package.json

```diff
@@ -3,7 +3,17 @@
   "version": "7.31.8",
   "author": "Yannick Croissant <yannick.croissant+npm@gmail.com>",
   "description": "React specific linting rules for ESLint",
-  "main": "index.js",
+  "main": "legacy.js",
+  "exports": {
+    ".": {
+      "import": "./index.js",
+      "require": "./legacy.js"
+    },
+    "./new": "./index.js",
```

Comment:
i'm a bit confused still. why is this different entrypoint needed?

Why can't the current "main" be used as-is in the new config system?

Reply:
The current "main" (which is renamed to legacy.js in this PR), does not follow the new config spec.
In the legacy system, a plugin can have preset(s) in the `configs` property.
However, plugin: protocol (e.g. `plugin:react/recommended`) is not valid any more in the new system. 
Thus, the new plugin(a new entrypoint index.js) doesn't have `configs`, following the spec.
This does not mean "the current main" cannot be used in the new system.
However, because it has a non-standard field `configs`, people might try to depend on it to load shareable configs, which is also not a standard way.
Eslint officially recommends people t o import shareable config directly, and this is an intention to establish the intended convention.

Reply:
I'm comfortable with the risk of people choosing to do that. Since the current plugin Just Works in the new config system, it seems like the only change that might be beneficial is adding new entry points for the configs to be accessed directly?